### PR TITLE
Configure dependabot to check for updates to dockerfiles

### PR DESCRIPTION
**Pull request by ethanmills**
## Why

To make it easier to maintain up-to-date dependencies

## What
Configure dependabot to raise update PRs against docker base images

## Technical writer support
None

## How to review
Basic syntax check

## Changelog
Not required

## Confirm

- [x] I have checked if any docs change here also requires updates to other repositories (ADRs / RFCs, README.md, Team Manual, elsewhere in these docs)
- [x] Where there is any overlap I have updated or opened a PR for corresponding changes

### CODE REVIEW

**Review by huwd — DISMISSED** [review]

**Comment by huwd** [comment]
Spoke to @andyloughran about this, @ethanmills you might wanna check that GitHub can get access to scan this properly.
Andy reckoned that apparently DepBot doesn't look at the static docker file it looks at the built artifact.

And if the registery isn't the standard docker.io registery then what depbot can do here might be limited by the access it has to AWS.... which _should_ be push only here as part of least privilidge.

@andyloughran has a plan here, so perhaps chat with him before hitting merge?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -11,3 +11,11 @@ updates:
     schedule:
       # Check for updates to GitHub Actions every week
       interval: "weekly"
+    commit-message:
+      prefix: "BAU"
```
Comment (andyloughran) [inline .github/dependabot.yml:15]: I don't think we should use BAU in official docs.  Our policy is that all changes should have a JIRA ticket; so can we change this.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -11,3 +11,11 @@ updates:
...
+  - package-ecosystem: docker
+    directory: "/"
+    schedule:
+      interval: daily
+    commit-message:
+      prefix: BAU
```
Comment (andyloughran) [inline .github/dependabot.yml:21]: See previous comment; we can't codify practices that we don't officially endorse.

**Comment by andyloughran** [comment]
I've reviewed by plan as part of LS-2262, and I think the only thing that's missing will be metadata regarding the deltas.  AWS::Inspector will look deeper inside the container that dependabot is able to; so I think this configuration recommendation is good for now.

However, I would remove references to 'BAU' - as this is not a practice our official documentation should endorse.  All changes should have an associated JIRA ticket.

**Comment by ethanmills** [comment]
BAUs removed.

**Review by andyloughran — APPROVED** [review]

**Pull request merged**